arcgis-ios-sdk-samples/Layers/Blend renderer/BlendRendererViewController.swift

```diff
@@ -19,91 +19,113 @@ import ArcGIS
-
-        //apply the blend renderer on this new raster layer
-        self.rasterLayer.renderer = blendRenderer
+        // create a blend renderer
+        let blendRenderer = AGSBlendRenderer(elevationRaster: elevationRaster,
+                                        outputMinValues: [9],
+                                        outputMaxValues: [255],
+                                        sourceMinValues: [],
+                                        sourceMaxValues: [],
+                                        noDataValues: [],
+                                        gammas: [],
+                                        colorRamp: colorRamp,
+                                        altitude: altitude,
+                                        azimuth: azimuth,
+                                        zFactor: 1,
+                                        slopeType: slopeType,
+                                        pixelSizeFactor: 1,
+                                        pixelSizePower: 1,
+                                        outputBitDepth: 8)
         
-        self.hideVisualEffectView()
-    }
-    
-    //MARK: - Actions
-    
-    func hideVisualEffectView() {
-        self.visualEffectView.isHidden = true
-    }
-    
-    @IBAction func editRendererAction() {
-        //show visual effect view
-        self.visualEffectView.isHidden = false
+        // remove the exisiting layers
+        mapView.map?.basemap.baseLayers.removeAllObjects()
+        
+        // if the colorRamp type is .none, then use the imagery for blending.
+        // else use the elevation
+        let rasterLayer = colorRampType == .none ? imageryBasemapLayer : elevationBasemapLayer
+        
+        // apply the blend renderer on this new raster layer
+        rasterLayer.renderer = blendRenderer
+
+        // add the raster layer to the basemap
+        mapView.map?.basemap.baseLayers.add(rasterLayer)
     }
-    
+
     //MARK: - Navigation
     
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        if segue.identifier == "SettingsEmbedSegue" {
-            let controller = segue.destination as! BlendRendererSettingsVC
+        if let controller = segue.destination as? BlendRendererSettingsVC,
+            let rasterLayer = mapView.map?.basemap.baseLayers.firstObject as? AGSRasterLayer,
+            let renderer = rasterLayer.renderer as? AGSBlendRenderer {
+           
+            controller.preferredContentSize = {
+                let height: CGFloat
+                if traitCollection.horizontalSizeClass == .regular,
+                    traitCollection.verticalSizeClass == .regular {
+                    height = 250
+                } else {
+                    height = 200
+                }
+                return CGSize(width: 375, height: height)
+            }()
+            controller.presentationController?.delegate = self
+            controller.loadViewIfNeeded()
```

Comment:
I see. The presenting view controller prevents the crash by loading the view first. I would argue that the settings view controller should Do The Right Thing™ and allow the caller to set these values before or after the view is loaded.